src/platform/linux_usbfs/enumeration.rs

```diff
@@ -122,6 +142,49 @@ pub fn list_devices() -> Result<impl Iterator<Item = DeviceInfo>, Error> {
     }))
 }
 
+pub fn list_root_hubs() -> Result<impl Iterator<Item = DeviceInfo>, Error> {
+    Ok(fs::read_dir(SYSFS_USB_PREFIX)?.filter_map(|entry| {
+        let path = entry.ok()?.path();
+        let name = path.file_name()?;
+
+        // root hubs are named `usbX` where X is the bus number
+        if !name.to_string_lossy().starts_with("usb") {
+            return None;
+        }
+
+        probe_device(SysfsPath(path))
+            .inspect_err(|e| warn!("{e}; ignoring root hub"))
+            .ok()
+    }))
+}
+
+pub fn list_buses() -> Result<impl Iterator<Item = BusInfo>, Error> {
+    Ok(list_root_hubs()?.filter_map(|rh| {
+        // get the parent by following the absolute symlink; root hub in /bus/usb is a symlink to a dir in parent bus
+        let parent_path = rh
+            .sysfs_path()
+            .canonicalize()
+            .ok()
+            .and_then(|p| p.parent().map(|p| SysfsPath(p.to_owned())))?;
+
+        debug!("Probing PCI device {:?}", parent_path.0);
```

Comment:
This debug message may not be accurate: the parent device isn't necessarily a PCI device.
```suggestion
        debug!("Probing parent device {:?}", parent_path.0);
```